drivers/spi/spi_nrfx_spim.c

```diff
@@ -639,7 +666,10 @@ static int spi_nrfx_init(const struct device *dev)
 		.wake_pin = NRF_DT_GPIOS_TO_PSEL_OR(SPIM(idx), wake_gpios,     \
 						    WAKE_PIN_NOT_USED),	       \
 	};								       \
-	BUILD_ASSERT(!DT_NODE_HAS_PROP(SPIM(idx), wake_gpios) ||	       \
+	BUILD_ASSERT(!SPIM_HAS_PROP(idx, memory_regions) ||		       \
+		     CONFIG_SPI_NRFX_RAM_BUFFER_SIZE != 0,		       \
```

Comment:
is it required that this option is enabled when `memory_regions` exists? Isn't there an option for user to provide buffers from allowed location?

Reply:
Good point. This is an unnecessary limitation. We should get rid of this assertion.

Reply:
I thought that it's worth checking but I don't have a strong opinion on that. Removed the check